scripts/build.py

```diff
@@ -1167,20 +1212,24 @@ def generate_build_directory():
             COPY_INPUT_DIRS[i], COPY_OUTPUT_DIRS[i], hashes)
     _execute_tasks(copy_tasks)
 
-    SOURCE_DIRS = [
-        ASSETS_DEV_DIR, [
-            EXTENSIONS_DIRNAMES_TO_DIRPATHS['dev_dir'],
-            EXTENSIONS_DIRNAMES_TO_DIRPATHS['compiled_js_dir']
-        ], [
-            TEMPLATES_CORE_DIRNAMES_TO_DIRPATHS['dev_dir'],
-            TEMPLATES_CORE_DIRNAMES_TO_DIRPATHS['compiled_js_dir']
-        ],
-        THIRD_PARTY_GENERATED_DEV_DIR]
-    # This is required since we need to compare build/extensions and
-    # build/templates against both the compiled js dir and dev dir.
-    SOURCE_DIRS_TYPE = [
-        'single_dir', 'list_of_dirs', 'list_of_dirs', 'single_dir']
-    _verify_build(SOURCE_DIRS, COPY_OUTPUT_DIRS, SOURCE_DIRS_TYPE, hashes)
+    _verify_hashes(COPY_OUTPUT_DIRS, hashes)
+
+    SOURCE_DIRS = [ASSETS_DEV_DIR, THIRD_PARTY_GENERATED_DEV_DIR]
+    OUTPUT_DIRS = [ASSETS_OUT_DIR, THIRD_PARTY_GENERATED_OUT_DIR]
+    _verify_file_count(SOURCE_DIRS, OUTPUT_DIRS)
+
+    SOURCE_DIRS_LIST = [[
```

Comment:
This is getting confusing TBH.

Why not call _verify_file_count twice and have it operate on two dir lists? Once you start doing lists of lists of dirs it becomes confusing as the lists in the middle layer don't have a good conceptual meaning.

Reply:
Updated - I split the source and output into four lists and called compare_file_count for each pair.